DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc

```diff
@@ -30,11 +20,10 @@ DDAlgorithmHandler::initialize( const std::string & algoName,
 				const DDStringArguments & sArgs,
 				const DDStringVectorArguments & svArgs )
 {
-  std::pair<std::string,std::string> algoNmNs = DDSplit(algoName);
   algoname_ = algoName;
-  algo_ = DDAlgorithmFactory::get()->create(algoname_);
-  algo_->setParent(parent);
-  algo_->initialize(nArgs,vArgs,mArgs,sArgs, svArgs);
+  algo_ = std::unique_ptr<DDAlgorithm>( DDAlgorithmFactory::get()->create( algoname_ ));
```

Comment:
The change isn't about performance. It is the recommended form to use from C++ style guides. If an exception happens during the construction phase, using the function will properly clean up everything.